harness/determined/pytorch/_callback.py

```diff
@@ -16,38 +16,33 @@ class PyTorchCallback:
 
     .. warning::
         If distributed training is enabled, every GPU will execute a copy of this callback
-        (except for :meth:`on_validation_step_end` and :meth:`on_checkpoint_end`).  To
-        configure a callback implementation to execute on a subset of GPUs, please condition
-        your implementation on ``trial.context.distributed.get_rank()``.
+        (except for :meth:`on_validation_end`, :meth:`on_validation_step_end` and
+        :meth:`on_checkpoint_end`). To configure a callback implementation to execute on a subset of
+        GPUs, please condition your implementation on ``trial.context.distributed.get_rank()``.
     """
 
-    def on_train_step_start(self, step_id: int) -> None:
+    def on_before_optimizer_step(self, parameters: Iterator) -> None:
         """
-        Run before every training step begins.
+        Run before every before ``optimizer.step()``.  For multi-GPU training, executes
+        after gradient updates have been communicated. Typically used to perform gradient
+        clipping.
         """
+        # TODO(DET-3267): deprecate this when releasing pytorch flexible primitives.
         pass
 
-    def on_train_step_end(self, step_id: int, metrics: Dict[str, Any]) -> None:
+    def on_validation_start(self) -> None:
         """
-        Run after every training step ends.
-
-        .. warning::
-            If distributed training is enabled, every GPU will execute a copy of
-            this callback at the end of every training step. If
-            ``optimizations.average_training_metrics`` is enabled, then the
-            ``metrics`` will be averaged across all GPUs before the callback is
-            executed.  If ``optimizations.average_training_metrics`` is
-            disabled, then the ``metrics`` will be local to the GPU.
+        Run before every validation begins.
         """
         pass
 
-    def on_before_optimizer_step(self, parameters: Iterator) -> None:
+    def on_validation_end(self, metrics: Dict[str, Any]) -> None:
         """
-        Run before every before ``optimizer.step()``.  For multi-GPU training, executes
-        after gradient updates have been communicated. Typically used to perform gradient
-        clipping.
+        Run after every validation ends.
+
+        .. warning::
+            This callback only executes on the chief GPU when doing distributed training.
         """
-        # TODO(DET-3267): deprecate this when releasing pytorch flexible primitives.
         pass
 
     def on_validation_step_start(self) -> None:
```

Comment:
non-blocking: might be worth adding a comment that this should be removed  in the future